tools/build_defs/framework.bzl

```diff
@@ -216,21 +217,35 @@ def cc_external_rule_impl(ctx, attrs):
 
     version_and_lib = "Bazel external C/C++ Rules #{}. Building library '{}'".format(VERSION, lib_name)
 
+    # We can not declare outputs of the action, which are in parent-child relashion,
+    # so we need to have a (symlinked) copy of the output directory to provide
+    # both the C/C++ artifacts - libraries, headers, and binaries,
+    # and the install directory as a whole (which is mostly nessesary for chained external builds).
+    #
+    # We want the install directory output of this rule to have the same name as the library,
+    # so symlink it under the same name but in a subdirectory
+    installdir_copy = copy_directory(ctx, "$INSTALLDIR", "copy_{}/{}".format(lib_name, lib_name))
+
+    # we need this fictive file in the root to get the path of the root in the script
+    empty = fictive_file_in_genroot(ctx)
+
+    define_variables = [
+        "export EXT_BUILD_ROOT=$BUILD_PWD",
+        "export BUILD_TMPDIR=$(mktemp -d)",
+        "export EXT_BUILD_DEPS=$EXT_BUILD_ROOT/bazel_foreign_cc_deps_" + lib_name,
+        "export INSTALLDIR=$EXT_BUILD_ROOT/" + empty.file.dirname + "/" + lib_name,
+    ]
+
     script_lines = [
         "echo \"\n{}\n\"".format(version_and_lib),
         "set -e",
         "source " + shell_utils,
         set_envs,
         "set_platform_env_vars",
-        "export EXT_BUILD_ROOT=$BUILD_PWD",
-        "export BUILD_TMPDIR=$(mktemp -d)",
-        "export EXT_BUILD_DEPS=$EXT_BUILD_ROOT/bazel_foreign_cc_deps_" + lib_name,
+        "\n".join(define_variables),
```

Comment:
This is actually a common part of the script, common for all OSes (provided we use it in Windows in Msys). No refactoring related to extracting to toolchain is needed. The whole framework.bzl is actually creating and calling this shell script.
And yep, of course I could do the moving of these three lines in the other PR, I agree.

Reply:
(Discussed offline)
Ok, this script should be refactored, since we will go away from requiring bash support for Windows.
(So more things should be changed like variables declaration etc)

Reverting this piece i this PR to keep changes more atomic.